src/build_interpret.ml

```diff
@@ -49,17 +49,23 @@ let inspect_path file_tree path =
       else
         None
 
+let no_targets_allowed () =
+  Exn.code_error "No targets allowed under a [Build.lazy_no_targets] \
+                  or [Build.if_file_exists]" []
+[@@inline never]
```

Comment:
Out of curiosity, what is the purpose of that annotation? is it just an optimization or is it required?

Reply:
It's just an optimization, so that the error case code is not inlined in the main function. It's kind of a _cold_  annotation.